datanommer.models/tests/test_model.py

```diff
@@ -243,6 +256,171 @@ def test_grep_not_topics(datanommer_models):
+
+    total, pages, messages = Message.grep(msg_id=bodhi_example_message.id)
+    assert total == 1
+    assert pages == 1
+    assert len(messages) == 1
+    assert messages[0].msg == bodhi_example_message.body
+
+    total, pages, messages = Message.grep(msg_id="NOTAMESSAGEID")
+    assert total == 0
+    assert pages == 0
+    assert len(messages) == 0
+
+
+def test_grep_users(datanommer_models):
+    example_message = generate_message()
+    add(example_message)
+
+    bodhi_example_message = generate_bodhi_update_complete_message()
+    add(bodhi_example_message)
+
+    session.flush()
+
+    total, pages, messages = Message.grep(users=["dudemcpants"])
+
+    assert total == 1
+    assert pages == 1
+    assert len(messages) == 1
+
+    assert messages[0].msg == bodhi_example_message.body
+
+
+def test_grep_not_users(datanommer_models):
+    example_message = generate_message()
+    add(example_message)
+
+    bodhi_example_message = generate_bodhi_update_complete_message()
+    add(bodhi_example_message)
+
+    session.flush()
+
+    total, pages, messages = Message.grep(not_users=["dudemcpants"])
+
+    assert total == 1
+    assert pages == 1
+    assert len(messages) == 1
+
+    assert messages[0].msg == example_message.body
+
+
+def test_grep_packages(datanommer_models):
+    example_message = generate_message()
+    add(example_message)
+
+    bodhi_example_message = generate_bodhi_update_complete_message()
+    add(bodhi_example_message)
+
+    session.flush()
+
+    total, pages, messages = Message.grep(packages=["kernel"])
+
+    assert total == 1
+    assert pages == 1
+    assert len(messages) == 1
+
+    assert messages[0].msg == bodhi_example_message.body
+
+
+def test_grep_not_packages(datanommer_models):
+    example_message = generate_message()
+    add(example_message)
+
+    bodhi_example_message = generate_bodhi_update_complete_message()
+    add(bodhi_example_message)
+
+    session.flush()
+
+    total, pages, messages = Message.grep(not_packages=["kernel"])
+
+    assert total == 1
+    assert pages == 1
+    assert len(messages) == 1
+
+    assert messages[0].msg == example_message.body
+
+
+def test_grep_rows_per_page_none(datanommer_models):
+    for x in range(0, 200):
+        example_message = generate_message()
+        example_message.id = f"{x}"
+        add(example_message)
+
+    session.flush()
+
+    total, pages, messages = Message.grep()
+    assert total == 200
+    assert pages == 2
+    assert len(messages) == 100
+
+    total, pages, messages = Message.grep(rows_per_page=None)
+    assert total == 200
+    assert pages == 1
+    assert len(messages) == 200
+
+
+def test_grep_defer(datanommer_models):
+    example_message = generate_message()
+    add(example_message)
+
+    session.flush()
+
+    total, pages, query = Message.grep(defer=True)
+    assert str(query.__class__) == "<class 'sqlalchemy.orm.query.Query'>"
+
+    assert query, all() == Message.grep()
```

Comment:
Wait what? What are you trying to tests here? The equality sign has precedence over the comma, so this lines means "assert that `query` is trueish, and if not print the evaluation of `all() == Message.grep()`." Since `all()` is the built-in function here, it'll fail because it requires at least one argument.
That's probably not what you want :-)

Reply:
yeah, this was supposed to be query.all() but i typoed a comma, and black fixed it up.

but i dont think it would have worked either.

will fix it up

Reply:
all i wanted to check was if the query returned when using defer returns the same results as when not using defer.

fix the test to do that properly now.